runtime/service/npud/core/ir/Layout.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONE_SERVICE_NPUD_CORE_IR_LAYOUT_H__
+#define __ONE_SERVICE_NPUD_CORE_IR_LAYOUT_H__
+
+#include <functional>
+#include <stdexcept>
+#include <string>
+
+namespace npud
+{
+namespace core
+{
+namespace ir
+{
+
+enum class Layout
+{
+  UNKNOWN = 0,
+  NHWC,
+  NCHW
+};
+
+inline std::string to_string(Layout layout)
+{
+  switch (layout)
+  {
+    case Layout::NHWC:
+      return std::string{"NHWC"};
+    case Layout::NCHW:
+      return std::string{"NCHW"};
+    case Layout::UNKNOWN:
+      return std::string{"UNKNOWN"};
+    default:
+      throw std::runtime_error("WRONG LAYOUT");
+  }
+}
+
+} // namespace ir
+} // namespace core
+} // namespace npud
+
+namespace std
+{
+
+template <> struct hash<npud::core::ir::Layout>
+{
+  size_t operator()(npud::core::ir::Layout value) const noexcept
+  {
+    using type = typename std::underlying_type<npud::core::ir::Layout>::type;
+    return hash<type>()(static_cast<type>(value));
+  }
+};
```

Comment:
Are you going to use `[Layout](npud::core::ir::Layout)` as hash key of map?

Reply:
Oh, not for now. :( I will remove it.